### ISSUE TITLE
[Improvement][DataSource] Whether it can split Hive/Impala DataSource Type to Hive Datasource Type and Impala Datasource Type ?

### ISSUE THREAD

**Issue by luna-learn** [apache/dolphinscheduler#15199]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

The protocol/port/driver of Hive Datasource is diffenernt fromImpala DataSource. 
For example:
- schema: The Hive protocol header is "jdbc:hive2://", but Impala protocol header is "jdbc:impala://".
- port: The Hive default port is 10000, but Impala default port is 21050.
- driver: The Hive jdbc driver is "org.apache.hive.jdbc.HiveDriver", but impala jdbc driver is "com.cloudera.impala.jdbc41.Driver".
To sum up, the Hive and Impala should not be mixed together. 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)
